src/Reflection/Annotation/AnnotationHelper.php

```diff
@@ -13,38 +13,21 @@ final class AnnotationHelper
 {
     public const TENTATIVE_RETURN_TYPE_ANNOTATION = 'betterReflectionTentativeReturnType';
 
-    private static ?DocBlockFactory $docBlockFactory = null;
-
-    /**
-     * @psalm-pure
-     */
     public static function isDeprecated(string $docComment): bool
     {
         if ($docComment === '') {
             return false;
         }
 
-        /** @psalm-suppress ImpureMethodCall */
-        return self::getDocBlockFactory()->create($docComment)->hasTag('deprecated');
+        return preg_match('~\*\s*@deprecated(?=\s|\*)~', $docComment) === 1;
     }
 
-    /**
-     * @psalm-pure
-     */
     public static function hasTentativeReturnType(string $docComment): bool
     {
         if ($docComment === '') {
             return false;
         }
 
-        /** @psalm-suppress ImpureMethodCall */
-        return self::getDocBlockFactory()->create($docComment)->hasTag(self::TENTATIVE_RETURN_TYPE_ANNOTATION);
-    }
-
-    private static function getDocBlockFactory(): DocBlockFactory
-    {
-        self::$docBlockFactory ??= DocBlockFactory::createInstance();
```

Comment:
As previously discussed in #899, I'm fully on board with dropping type parsing: what's the alternative these days, if we need to endorse it to end-users? 

Related: https://github.com/CuyZ/Valinor/issues/6
Related: https://github.com/phpstan/phpstan-src/tree/447cd102d946a2d9883ac82793195275f54296a9/src/Type
Related: https://github.com/vimeo/psalm/tree/f1d47cc662500589e200a978e3af85488a5236cf/src/Psalm/Type
Related: https://github.com/phpstan/phpdoc-parser
Related: https://github.com/phpstan/phpdoc-parser/blob/master/tests/PHPStan/Parser/TypeParserTest.php#L64

Overall, I'd like to at least explain (in the docs) that this library "ends" at providing docblocks, and that parsing them should be deferred to third-party libraries (since types are ever evolving there)

Reply:
phpdoc-parser is really low-level, good comparison is nikic/PHP-Parser. In order to do something useful, you'd need parts of PHPStan that are hard to separate, like reflection layer, and type system.

Reply:
But I acknowledge that phpdoc-parser can sometimes be useful on its own, for example Slevomat CS uses it :) But it needs to interpret the AST on its own.